cypress/pageobjects/image.po.ts

```diff
@@ -1,45 +1,183 @@
-    cy.get('.cru-resource-footer').contains('Create').click()
+  public exportImage(vmName: string, imageName: string) {
+    volumes.goToList();
+    cy.get('.search').type(vmName);
+
+    const volume = cy.contains(vmName);
+
+    volume.should('be.visible')
+
+    volume.parentsUntil('tbody', 'tr').find('td').eq(2).invoke('text').then((volumeName) => {
+      volumes.exportImage(volumeName.trim(), imageName);
+    });
+  }
+
+  public checkState(value: ValueInterface, valid: boolean = true) {
+    cy.get('.search').type(value.name);
+    cy.contains(value.name).parentsUntil('tbody', 'tr').find('td.col-image-percentage-bar').contains(valid ? 'Completed' : '0%', { timeout: 300000 }).should('be.visible');
+    cy.contains(value.name).parentsUntil('tbody', 'tr').find('td.col-badge-state-formatter').contains(valid ? 'Active' : 'Failed', { timeout: 300000 }).should('be.visible');
+
+    if (valid) {
+      cy.contains(value.size || 'Size need to be string').should('be.visible');
+    }
+  }
+
+  // public checkImageInLH(imageName: string) {
+  //   let imageId: any = null;
+
+  //   cy.intercept('GET', '/v1/harvester/harvesterhci.io.virtualmachineimages').as('getImageList');
+  //   this.goToList();
+
+  //   cy.wait('@getImageList').then(res => {
+  //     expect(res.response?.statusCode).to.equal(200);
+
+  //     const images = res.response?.body?.data || [];
+
+  //     imageId = images.find((image: any) => image.spec.displayName === imageName)?.id;
+  //     expect(!!imageId).to.be.true;
+  //   });
+
+  //   cy.request({ method: 'GET', url: 'v1/longhorn.io.backingimages', headers: {
+  //     'Content-Type': 'application/json; charset=utf-8',
+  //     'accept': 'json'
+  //   }}).as('getBackingImageList');
+
+  //   cy.get('@getBackingImageList').should((response) => {
+  //     const found = response.body.data.find((image: any) => imageId.replace('/', '-') === image.metadata.name);
+
+  //     expect(found?.metadata?.state?.error).to.be.false;
+  //     expect(found?.metadata?.state?.name).to.equal('active');
+  //   });
+  // }
+
+  public save( { upload, edit }: { upload?: boolean; edit?: boolean } = {} ) {
+    cy.intercept(edit? 'PUT' : 'POST', `/v1/harvester/harvesterhci.io.virtualmachineimages${ upload ? '/*' : edit ? '/*/*' : '' }`).as('createImage');
+    cy.get('.cru-resource-footer').contains(!edit ? 'Create' : 'Save').click();
     cy.wait('@createImage').then(res => {
-      expect(res.response?.statusCode).to.equal(201);
-    })
+      expect(res.response?.statusCode).to.equal( edit ? 200 : 201 );
+    });
   }
 
   public setValue(value: ValueInterface) {
-    this.url().input(value?.name)
+    this.name().input(value?.name);
+    this.url().input(value?.url);
+    this.description().input(value?.description);
   }
 
-  url() {
-    return new LabeledInputPo('.labeled-input', `:contains("URL")`)
+  public delete(name: string) {
+    this.goToList();
+    cy.get('.search').type(name);
```

Comment:
The selector should be stored as a class member not selected just based on a CSS class
```
private search = '.search';
cy.get(this.search).type(name);


```

Reply:
done